pkg/common/upgrade/managed.go

```diff
@@ -0,0 +1,268 @@
+		return fmt.Errorf("can't parse upgrade config override template: %v", err)
+	}
+
+	// only update if we observe a difference between current config and testing config
+	if cfgData != configOverride {
+		cm.Data["config.yaml"] = configOverride
+		_, err = h.Kube().CoreV1().ConfigMaps(muoNamespace).Update(context.TODO(), cm, metav1.UpdateOptions{})
+		if err != nil {
+			return fmt.Errorf("managed-upgrade-operator configmap could not be updated: %v", err)
+		}
+	}
+
+	return nil
+}
+
+// Create the UpgradeConfig CR for the managed-upgrade-operator which will
+// cause the upgrade process to initiate.
+func createUpgradeConfig(channel string, version string, h *helper.H) error {
+
+	// Delete any existing UpgradeConfig
+	h.Dynamic().Resource(schema.GroupVersionResource{
+		Group: "upgrade.managed.openshift.io", Version: "v1alpha1", Resource: "upgradeconfigs",
+	}).Namespace(muoNamespace).Delete(context.TODO(), upgradeConfigName, metav1.DeleteOptions{})
+
+	// Create a new UpgradeConfig and add it to the cluster
+	uc := upgradev1alpha1.UpgradeConfig{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "UpgradeConfig",
+			APIVersion: "upgrade.managed.openshift.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      upgradeConfigName,
+			Namespace: muoNamespace,
+		},
+		Spec: upgradev1alpha1.UpgradeConfigSpec{
+			Desired: upgradev1alpha1.Update{
+				Version: version,
+				Channel: channel,
+			},
+			UpgradeAt:            time.Now().UTC().Format(time.RFC3339),
+			PDBForceDrainTimeout: muoPdbDrainTimeout,
+			Type:                 muoUpgradeType,
+		},
+	}
+
+	obj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(uc.DeepCopy())
+	if err != nil {
+		return fmt.Errorf("can't convert UpgradeConfig to unstructured resource: %v", err)
+	}
+
+	uobj := unstructured.Unstructured{obj}
+	_, err = h.Dynamic().Resource(schema.GroupVersionResource{
+		Group:    "upgrade.managed.openshift.io",
+		Version:  "v1alpha1",
+		Resource: "upgradeconfigs",
+	}).Namespace(muoNamespace).Create(context.TODO(), &uobj, metav1.CreateOptions{})
+
+	if err != nil {
+		return fmt.Errorf("can't create UpgradeConfig resource: %v", err)
+	}
+
+	return nil
+}
+
+// creates workloads that are protected by Pod Disruption Budgets (PDBs)
+// which the managed-upgrade-operator is expected to be able to handle
+func createPodDisruptionBudgetWorkloads(h *helper.H) error {
+
+	if _, ok := h.GetWorkload(pdbWorkloadName); ok {
+		return nil
+	}
+
+	// Create all K8s objects that are within the testDir
+	log.Printf("Applying Pod Disruption Budget workload: %s\n", pdbWorkloadDir)
+	_, err := helper.ApplyYamlInFolder(pdbWorkloadDir, h.CurrentProject(), h.Kube())
+	if err != nil {
+		return fmt.Errorf("can't create upgrade PDB workload: %v", err)
+	}
+
+	// Give the cluster a second to churn before checking
+	time.Sleep(3 * time.Second)
+
+	// Wait for all pods to come up healthy
+	err = wait.PollImmediate(15*time.Second, 10*time.Minute, func() (bool, error) {
```

Comment:
Could this timeout be lessened? I'm starting to get concerned with the amount of 10min timeouts that we're accumulating. :) 

Also, should this be MaxDuration instead of a static 10min?

Reply:
Oh absolutely, sorry about that - that's me being lazy and copy/pasting an existing block without putting any thought into the timings. I've tightened it up quite a bit in the latest push.